### PR TITLE
Fix coming soon page mobile UI issues

### PR DESCRIPTION
**Pull request by chihsuan**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

p1715675706047389-slack-C01SFMVEYAK

Fix coming soon page not displaying correctly on mobile view

- Add min-width to body tag to prevent the page from being too small on mobile view
- Adjust padding for smaller screens
- Update font size and text wrapping for banner. Even after adding `text-wrap: balance` or `&nbsp;`, the window issue still exists with the banner text with Samsung Galaxy S8+ because the font size was really too large. So, I referred to the editor and made the necessary changes (use [clamp()](https://developer.mozilla.org/en-US/docs/Web/CSS/clamp)) to the font size to make it look better on mobile view.

Both `text-wrap: balance` and `&nbsp;` should work for the banner text. But I used `text-wrap: balance` in the PR because I found we used `text-wrap` in WordPress.com p7H4VZ-4Hm-p2#comment-11820 and seems it's a better way to handle the text wrapping issue. Besides, I think that would still prevent the text from being cut off if users change the banner text

Before:

### Samsung Galaxy S8+ - without font size adjustment

Screenshot 2024-05-15 at 11 33 11

### Samsung Galaxy S8+ - with font size adjustment

Screenshot 2024-05-15 at 11 58 42

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a fresh site with `launch-your-store` feature flag enabled.
2. Skip the setup wizard
3. Go to frontend in incognito mode
4. Check the coming soon page looks good on mobile view

https://github.com/woocommerce/woocommerce/assets/4344253/c58bc215-9f56-4d06-a496-75838d360e08

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-blocks/assets/js/blocks/coming-soon/styles.tsx [hunk 1]
```diff
@@ -129,7 +135,7 @@ export const generateStyles = ( color = '#bea0f2' ) => {
         height: 40px;
     }
     .woocommerce-coming-soon-banner {
-        font-size: 48px;
+        font-size: clamp(27.894px, 1.743rem + ((1vw - 3.2px) * 2.094), 48px);
```
Comment (chihsuan) [inline plugins/woocommerce-blocks/assets/js/blocks/coming-soon/styles.tsx]: I set the font size to `48px` in the site editor and copied-pasted this value from the applied CSS.

The syntax for `clamp()` is:

```css
clamp(minimum, preferred, maximum)
```



- Minimum Value: 27.894px
- Preferred Value: 1.743rem + ((1vw - 3.2px) * 2.094)
- Maximum Value: 48px

Comment (adrianduffell) [reply plugins/woocommerce-blocks/assets/js/blocks/coming-soon/styles.tsx]: Nitpick: It sounds like you copied the values out of the editor, but perhaps we should round the numbers to ease readability 🤔

**Review by adrianduffell — APPROVED** [review]
Thanks @chihsuan, this is testing great and LGTM. I tried it in various different mobile and tablet sizes. I just left a nitpick but feel free to ignore 🚀

File: plugins/woocommerce-blocks/assets/js/blocks/coming-soon/styles.tsx (hunk 1 above)
Comment (chihsuan) [reply plugins/woocommerce-blocks/assets/js/blocks/coming-soon/styles.tsx]: Sounds good! Thanks for the suggestion.

Comment (chihsuan) [reply plugins/woocommerce-blocks/assets/js/blocks/coming-soon/styles.tsx]: Updated in https://github.com/woocommerce/woocommerce/pull/47491/commits/f6bd20e5c3ee194694fd527948853c748a21cdeb

**Review by ilyasfoo — APPROVED** [review]
Tests well, LGTM! 🚢

**Pull request merged**